modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/io/Sinks.scala

```diff
@@ -83,14 +81,22 @@ object Sinks {
 
     GooglePubsubProducer
       .of[F, A](ProjectId(output.project), Topic(output.name), config)
-      .map(producer =>
-        (s: Stream[F, AttributedData[A]]) =>
-          s.parEvalMapUnordered(Enrich.ConcurrencyLevel)(row => producer.produce(row.data, row.attributes).void)
-      )
+      .map { producer => row: AttributedData[A] => producer.produce(row.data, row.attributes).void }
   }
 
-  private def fileSink[F[_]: Sync: ContextShift](path: Path, blocker: Blocker): ByteSink[F] =
-    _.flatMap(bytes => Stream.emits(bytes) ++ Stream.emit('\n'.toByte))
-      .through(writeAll[F](path, blocker, List(StandardOpenOption.CREATE_NEW)))
+  private def fileSink[F[_]: Concurrent: ContextShift](path: Path, blocker: Blocker): Resource[F, ByteSink[F]] =
```

Comment:
I find it a bit disturbing to have a `Resource` of a function, although I agree that it's very convenient.